packages/applicant/src/middleware.page.ts

```diff
@@ -195,26 +150,115 @@ async function shouldRedirectToClosedGrantPage(jwt: string, req: NextRequest) {
   }
 }
 
-const newApplicationPattern = new URLPattern({
-  pathname: '/applications/:applicationId([0-9]+)',
-});
+const authenticateRequest = async (req: NextRequest, res: NextResponse) => {
+  if (patterns.signInDetails.test({ pathname: req.nextUrl.pathname })) {
+    if (!ONE_LOGIN_ENABLED) {
+      const url = req.nextUrl.clone();
+      url.pathname = '/404';
+      return NextResponse.redirect(url);
+    } else {
+      return NextResponse.next();
+    }
+  }
 
-const redirectFromFindPattern = new URLPattern({
-  pathname: '/api/redirect-from-find',
-});
+  const jwt = await getJwtFromMiddlewareCookies(req);
 
-const signInDetailsPage = new URLPattern({
-  pathname: '/sign-in-details',
-});
+  console.log({ jwt });
 
-const mandatoryQuestionsStartPattern = new URLPattern({
-  pathname: '/mandatory-questions/start',
-});
+  if (
+    patterns.submissionJourney.test({ pathname: req.nextUrl.pathname }) ||
+    patterns.mandatoryQuestionsJourney.test({ pathname: req.nextUrl.pathname })
+  ) {
+    const shouldRedirect = await shouldRedirectToClosedGrantPage(jwt, req);
+    if (shouldRedirect) {
+      return shouldRedirect;
+    }
+  }
 
-const mandatoryQuestionsJourneyPattern = new URLPattern({
-  pathname: '/mandatory-questions/:questionUuid([0-9a-f-]+)/:path*',
-});
+  const validJwtResponse = await verifyToken(jwt);
+  const expiresAt = new Date(validJwtResponse.expiresAt);
+  console.log(validJwtResponse);
+  if (!validJwtResponse.valid) {
+    console.log('jwt invalid');
+    return buildMiddlewareResponse(req, HOST);
+  }
+
+  if (isWithinNumberOfMinsOfExpiry(expiresAt, 30)) {
+    return buildMiddlewareResponse(
+      req,
+      `${process.env.REFRESH_URL}?redirectUrl=${process.env.HOST}${
+        req.nextUrl.pathname
+      }?${req.nextUrl.searchParams.toString()}`
+    );
+  }
+  return res;
+};
+
+const httpLoggers = {
+  req: (req: NextRequest) => {
+    const correlationId = v4();
+    req.headers.set(HEADERS.CORRELATION_ID, correlationId);
+    logger.http('Incoming request', {
+      ...logger.utils.formatRequest(req),
+      correlationId,
+    });
+  },
+  res: (req: NextRequest, res: NextResponse) =>
+    logger.http(
+      'Outgoing response - PLEASE NOTE: this represents a snapshot of the response as it exits the middleware, changes made by other server code (eg getServerSideProps) will not be shown',
+      {
+        ...logger.utils.formatResponse(res),
+        correlationId: req.headers.get(HEADERS.CORRELATION_ID),
+      }
+    ),
+};
 
-const submissionJourneyPattern = new URLPattern({
-  pathname: '/submissions/:path*',
-});
+type LoggerType = keyof typeof httpLoggers;
+
+const urlsToSkip = ['/_next/', '/assets/', '/javascript/'];
+
+const getConditionalLogger = (req, type: LoggerType) => {
+  const userAgentHeader = req.headers.get('user-agent') || '';
+  return userAgentHeader.startsWith('ELB-HealthChecker') ||
```

Comment:
In the case where `userAgentHeader.startsWith('ELB-HealthChecker')` returns `true` instead of a function - we going to throw an error when we run `logRequest(req, res);` in the middleware.  Should that scenario return `() => undefined` instead?